### PR TITLE
LPS-138925 Semver

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java [hunk 1]
```diff
@@ -122,6 +129,42 @@ public AddTaskRequest createAddTaskRequest(
...
+		).name(
+			kaleoTransition.getName()
+		).nodeId(
+			_getNodeId(kaleoTransition.getKaleoNodeId())
+		).processId(
+			kaleoTransition.getKaleoDefinitionId()
+		).processVersion(
+			processVersion
+		).sourceNodeId(
+			_getNodeId(kaleoTransition.getSourceKaleoNodeId())
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java:155]: Hi @seltong,

Create another sourceNodeId method which a UnsafeSupplier as parameter.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TransitionWorkflowMetricsIndexerImpl.java [hunk 2]
```diff
@@ -81,6 +79,46 @@ public Document addTransition(
...
 	}
 
+	@Override
+	public Document addTransition(
+		long companyId, Date createDate, Date modifiedDate, String name,
+		long nodeId, long processId, String processVersion, long sourceNodeId,
+		String sourceNodeName, long targetNodeId, String targetNodeName,
+		long transitionId, long userId) {
+
+		AddTransitionRequest.Builder builder =
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TransitionWorkflowMetricsIndexerImpl.java:95]: @seltong, please rename this to addTransitionRequestBuilder.

thanks!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TransitionWorkflowMetricsIndexerImpl.java [hunk 3]
```diff
@@ -31,6 +32,53 @@
...
+		).setString(
+			"uid",
+			digest(
+				addTransitionRequest.getCompanyId(),
+				addTransitionRequest.getTransitionId())
+		).setLong(
+			"userId", addTransitionRequest.getUserId()
+		).setString(
+			"version", addTransitionRequest.getProcessVersion()
+		);
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-service/src/main/java/com/liferay/portal/workflow/metrics/internal/search/index/TransitionWorkflowMetricsIndexerImpl.java:72]: Hi @seltong, 

put this block inline, like this `addDocument(...).build())`

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java [hunk 4]
```diff
@@ -218,6 +219,19 @@ public AddTransitionRequest createAddTransitionRequest(
 		return localizationMap;
 	}
 
+	public DeleteTransitionRequest createDeleteTransitionRequest(
+		KaleoTransition kaleoTransition) {
+
+		DeleteTransitionRequest.Builder builder =
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java:225]: @seltong, rename this variable to deleteTransitionRequestBuilder.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/search/index/TransitionWorkflowMetricsIndexer.java [hunk 5]
```diff
@@ -25,12 +25,8 @@
 
 	public Document addTransition(AddTransitionRequest addTransitionRequest);
 
-	public default void deleteTransition(
-		DeleteTransitionRequest deleteTransitionRequest) {
-
-		throw new UnsupportedOperationException(
-			"This method needs to be implemented");
-	}
+	public void deleteTransition(
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/search/index/TransitionWorkflowMetricsIndexer.java:28]: @seltong, move this change to the creational commit.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
ci:test:sf

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Pull request closed without merging**